README.md

```diff
@@ -1 +1,130 @@
-# earthstar-fetch
\ No newline at end of file
+# earthstar-fetch
+
+A `fetch()` interface for talking to EarthStar
+
+## Goals
+
+- Stick to existing conventions in HTTP-land (body, methods, prefer standard headers)
+- GET data from earthstar workspaces
+- GET an author address for a given name
+- PUT data into a workspace
+- Auto-generate identity keys from a 4 character name
+- Search via querystring params and path prefixes
+
+## Dream API
+
+Basic URL format:
+
+```
+earthstar://+gardening.xxxxxxxxx/path/to/key.json
+```
+
+GET document
+
+```JavaScript
+const res = await fetch('earthstar://+gardening.xxxxx/example')
```

Comment:
To delete a doc in Earthstar you overwrite it with an empty string.  So empty docs should be considered to not exist even though they have metadata.

So this should 404 if the requested path doesn't exist, or if the content is an empty string.

Also to consider: In the future Earthstar will have a sparse mode where content can be `null`, meaning not downloaded yet, but the rest of the metadata will be there.  You might fetch the content on demand.

Reply:
Good call, I'll add an example.

Also we should be able to do a `HEAD` request for when we want to get the document metdata without the content.

Reply:
Yeah, we do still need to be able to get the metadata even if the content is empty.

I think... if we're asking for content, then empty content should come back as a 404 as a special case.  If we're asking for the whole document or the metadata, it should come back with an actual answer.

`HEAD` seems an obvious way to just get metadata.  If we also do the format I describe in the next comment (`/foo?kind=metadata`) which returns JSON, then maybe the `HEAD` would just return the metadata as HTTP headers if you're excited about that?

Reply:
HEAD shouldn't have a body so I think just the headers could be good. Do we need a metadata string when we can get the entire document?